amplify/backend/ts-code/src/api/AddItem.ts

```diff
@@ -68,6 +68,8 @@ export class AddItem {
         }
     }
 
+    private getUniqueId = Math.floor((Math.random() * Date.now() * new Date().getMilliseconds()) % 10).toString(36).substring(0, 6)
```

Comment:
Can you make it into a function:
```
private getUniqueId(): string {
   return Math.floor((Math.random() * Date.now() * new Date().getMilliseconds()) % 10).toString(36).substring(0, 6)
}
```

Also since you're multiplying the date by a random factor, I think you need to add a check for duplicates. You can make the function recurring in that it generates the id, and then checks if the id exists in the `items` table. If it doesn't exist, then return it, otherwise, call itself again.

Reply:
You can see [here](https://github.com/gp-seattle/rms-app/blob/master/amplify/backend/ts-code/src/db/ItemTable.ts#L21-L24) how I checked for duplicates.